wyvern/experimentation/client.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+import logging
+from typing import Optional
+
+from wyvern.config import settings
+from wyvern.exceptions import ExperimentationProviderNotSupportedError
+from wyvern.experimentation.providers.base import ExperimentationProvider
+from wyvern.experimentation.providers.eppo_provider import EppoExperimentationClient
+
+logger = logging.getLogger(__name__)
+
+
+class ExperimentationClient:
+    def __init__(self, provider_name: str):
+        """
+        Initializes the ExperimentationClient with a specified provider.
+
+        Args:
+        - provider_name (str): The name of the experimentation provider (e.g., "eppo").
+        """
+        if not settings.EXPERIMENTATION_ENABLED:
+            logger.info("Experimentation is disabled")
+            self.enabled = False
+            return
+
+        self.enabled = True
+        if provider_name == ExperimentationProvider.EPPO:
+            logger.info("Using EPPO experimentation provider")
+            self.provider = EppoExperimentationClient()
+        else:
+            raise ExperimentationProviderNotSupportedError(provider_name=provider_name)
+
+    def get_experiment_result(
+        self, experiment_id: str, entity_id: str, **kwargs
+    ) -> Optional[str]:
+        """
+        Get the result (variant) for a given experiment and entity using the chosen provider.
+
+        Args:
+        - experiment_id (str): The unique ID of the experiment.
+        - entity_id (str): The unique ID of the entity.
+        - kwargs (dict): Any additional arguments to pass to the provider for targeting.
+
+        Returns:
+        - str: The result (variant) assigned to the entity for the specified experiment.
+        """
+        if not self.enabled:
+            logger.error(
+                "get_experiment_result called when experimentation is disabled",
+            )
+            return None
+
+        result = None
+        error_message = None
+
+        try:
+            result = self.provider.get_result(experiment_id, entity_id, **kwargs)
+        except Exception as e:
+            error_message = str(e)
+
+        self.provider.log_result(
+            experiment_id, entity_id, result, error_message, **kwargs
```

Comment:
I don't think we should log the raw error message or at least that's not enough.    this experimentation log here is more for data scientists to understand what happened with sql query. it might be good for engineering debugging but not for DS to run some.

i think we should have some attributes that's query friendly for "why we're getting a None value", whether it's 1) provider server issue.2) not in the experimentation 3) or more.

Reply:
How do you think we can do it?

Reply:
Checked the sdk implementation and there isn't a way to separate sdk errors from wyvern errors. 

instead I'm going to convert the field to a boolean to indicate that it's an erronous result and I'll just log the exception message for debugging purposes.

Reply:
SGTM